<launch>
    <!-- Needed launch arguments -->
    <arg name="bal_dataset_path"
         default="$(find simple_sfm_ros)/data/problem-21-11315-pre.txt"
         doc="The path to the BAL dataset file." />
    <arg name="save_csvs"
         default="true"
         doc="Flag to indicate whether to save CSV files." />
    <arg name="output_csv_folder_path"
         default="$(find simple_sfm_ros)/data/"
         doc="The path to the folder where the CSV files will be saved." />
    <arg name="launch_rviz"
         default="false"
         doc="Flag to indicate whether to launch RViz visualization." />

    <!-- Publishing a static transform to make it easier to look bal points in rviz. -->
    <!-- bal world frame rotation w.r.t. map frame: yaw=-90, roll=90 degrees -->
    <node
        pkg="tf"
        type="static_transform_publisher"
        name="bal_to_map_tf_br"
        args="0 0 0 -1.57 0 1.57 map bal 100"
    />

    <!-- simple_sfm_node -->
    <node
        ns="simple_sfm"
        pkg="simple_sfm_ros"
        name="simple_sfm_node"
        type="simple_sfm_node"
        output="screen">
            <!-- Forward args as the node's private params -->
            <param name="bal_dataset_path" value="$(arg bal_dataset_path)" />
            <param name="save_csvs" value="$(arg save_csvs)" />
            <param name="output_csv_folder_path" value="$(arg output_csv_folder_path)" />
    </node>

    <!-- rviz -->
    <node
        if="$(arg launch_rviz)"
        pkg="rviz"
        type="rviz"
        name="simple_sfm_rviz"
        args="-d $(find simple_sfm_ros)/rviz/simple_sfm.rviz"
    />
</launch>
